src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -160,4 +204,28 @@ public void setToNewMatches(CaseData data, List<CaseMatch> newMatches) {
             data.setCaseMatches(null);
         }
     }
+
+    List<uk.gov.hmcts.reform.ccd.client.model.CaseDetails> getOldDivorceFreshMatches(MarriageDetails marriageDetails) {
+        BoolQueryBuilder nameMatchQuery1 = QueryBuilders.boolQuery()
+            .filter(QueryBuilders.termQuery("data.D8MarriageRespondentName.keyword", marriageDetails.getApplicant1Name()))
+            .filter(QueryBuilders.termQuery("data.D8MarriagePetitionerName.keyword", marriageDetails.getApplicant2Name()));
```

Comment:
Do you know whether the name matching should be case-insensitive?

Reply:
it shouldn't be

Reply:
I'll change to matchQuery

Reply:
actually, no, cos it should be exact match, I'll think of something else

Reply:
hm, we have lowercase normalizer on our elastic, I'll need to check the one for old Divorce has it to be sure

Reply:
I meant it should be case-insensitive, I misread your comment.
I checked in AAT, it picks up a case that's got caps at start of name in nfd and matches with a case that's lower case for whole name in old divorce, so it must be same lowercase es mapping